### PR TITLE
DDF-2652 updated OSGi basics docs

### PR DESCRIPTION
**Pull request by ricklarsen**
#### What does this PR do?

Updates documentation for OSGi basics

#### Who is reviewing it?

@mcalcote @vinamartin @Lambeaux @jrnorth @tbatie @adimka @garrettfreibott @djblue @peterhuffer 

#### Select at least one member from relevant component team(s) from below (at least one component team member needs to approve the PR).

[Docs](https://github.com/orgs/codice/teams/docs)
[Core APIs](https://github.com/orgs/codice/teams/core-apis)

#### Choose 2 committers to review/merge the PR (please choose ONLY two committers from below, delete the rest).

@coyotesqrl
@lessarderic
@pklinef
@shaundmorris
@kcwire 

#### How should this be tested? (List steps with links to updated documentation)

review revised content
#### Any background context you want to provide?
n/a

#### What are the relevant tickets?
[DDF-2652](https://codice.atlassian.net/browse/DDF-2652)

#### Screenshots (if appropriate)

[ddf-2652.zip](https://github.com/codice/ddf/files/862125/ddf-2652.zip)

#### Checklist:
- [x] Documentation Updated
- [ ] Change Log Updated
- [ ] Update / Add Unit Tests
- [ ] Update / Add Integration Tests

### CODE REVIEW

File: distribution/docs/src/main/resources/_contents/_developing/developing-intro-contents.adoc [hunk 1]
```diff
@@ -170,100 +170,106 @@ ${ddf-branding-lowercase}${at-symbol}local>list | grep -i example
...
-    <configfile finalname="/data/solr/metacard_cache/conf/schema.xml">mvn:ddf.platform.solr/platform-solr-server-standalone/${ddf.version}/xml/schema</configfile>
-    <configfile finalname="/data/solr/metacard_cache/conf/protwords.txt">mvn:ddf.platform.solr/platform-solr-server-standalone/${ddf.version}/txt/protwords</configfile>
-    <configfile finalname="/data/solr/metacard_cache/conf/stopwords_en.txt">mvn:ddf.platform.solr/platform-solr-server-standalone/${ddf.version}/txt/stopwords_en</configfile>
-    <configfile finalname="/data/solr/metacard_cache/conf/stopwords.txt">mvn:ddf.platform.solr/platform-solr-server-standalone/${ddf.version}/txt/stopwords</configfile>
-    <configfile finalname="/data/solr/metacard_cache/conf/synonyms.txt">mvn:ddf.platform.solr/platform-solr-server-standalone/${ddf.version}/txt/synonyms</configfile>
-</feature>
+<features name="${variable-prefix}project.artifactId}-${variable-prefix}project.version}">
+
+    <feature name="calculator" install="manual" version="${variable-prefix}project.version}"
```
Comment (kcwire) [inline distribution/docs/src/main/resources/_contents/_developing/developing-intro-contents.adoc]: This is a really big example with very little context. I'd recommend using the example from below and changing the flow to talk through the different aspects of the features file, which I believe is done below.

Comment (ricklarsen) [inline distribution/docs/src/main/resources/_contents/_developing/developing-intro-contents.adoc]: removed.

File: distribution/docs/src/main/resources/_contents/_developing/developing-intro-contents.adoc [hunk 2]
```diff
@@ -170,100 +170,52 @@ ${ddf-branding-lowercase}${at-symbol}local>list | grep -i example
...
 ----
-<feature name="catalog-app" install="auto">
-    <feature>platform-app</feature>
-    <feature>catalog-core</feature>
-    <feature>catalog-core-metricsplugin</feature>
-    <feature>catalog-core-sourcemetricsplugin</feature>
-    <feature>catalog-transformer-thumbnail</feature>
+<feature name="example-app" install="auto">
+    <feature prerequisite="true">prereq-feature</feature>
+    <bundle>mvn:com.example/example/dependency</bundle>
```
Comment (coyotesqrl) [inline distribution/docs/src/main/resources/_contents/_developing/developing-intro-contents.adoc]: The format for your sample `<bundle>`s is wrong. It should be
`mvn:[group name]/[artifact name]/[version][/[optional classifier]]`

e.g.
`<bundle>mvn:com.example/example.dependency/1.3</bundle>`

or in the less common case where a classifier is specified:
`<bundle>mvn:com.example/example.dependency/1.3/war</bundle>`

(Assuming an artifact name of `example.dependency`.)

Comment (ricklarsen) [inline distribution/docs/src/main/resources/_contents/_developing/developing-intro-contents.adoc]: fixed.

**Pull request merged**